ts/Series/Pie/PiePoint.ts

```diff
@@ -0,0 +1,430 @@
+                typeof vis === 'undefined' ? !point.visible : vis;
+            // update userOptions.data
+            (series.options.data as any)[series.data.indexOf(point)] =
+                point.options;
+
+            // Show and hide associated elements. This is performed
+            // regardless of redraw or not, because chart.redraw only
+            // handles full series.
+            ['graphic', 'dataLabel', 'connector', 'shadowGroup'].forEach(
+                function (key: string): void {
+                    if ((point as any)[key]) {
+                        (point as any)[key][vis ? 'show' : 'hide'](true);
+                    }
+                }
+            );
+
+            if (point.legendItem) {
+                chart.legend.colorizeItem(point, vis);
+            }
+
+            // #4170, hide halo after hiding point
+            if (!vis && point.state === 'hover') {
+                point.setState('');
+            }
+
+            // Handle ignore hidden slices
+            if (ignoreHiddenPoint) {
+                series.isDirty = true;
+            }
+
+            if (redraw) {
+                chart.redraw();
+            }
+        }
+    }
+
+    /**
+     * Set or toggle whether the slice is cut out from the pie.
+     * @private
+     *
+     * @param {boolean} sliced
+     * When undefined, the slice state is toggled.
+     *
+     * @param {boolean} redraw
+     * Whether to redraw the chart. True by default.
+     *
+     * @param {boolean|Partial<Highcharts.AnimationOptionsObject>}
+     * Animation options.
+     */
+    public slice(
+        sliced: boolean,
+        redraw?: boolean,
+        animation?: (boolean|Partial<AnimationOptionsObject>)
+    ): void {
+        var point = this,
+            series = point.series,
+            chart = series.chart;
+
+        setAnimation(animation, chart);
+
+        // redraw is true by default
+        redraw = pick(redraw, true);
+
+        /**
+         * Pie series only. Whether to display a slice offset from the
+         * center.
+         * @name Highcharts.Point#sliced
+         * @type {boolean|undefined}
+         */
+        // if called without an argument, toggle
+        point.sliced = point.options.sliced = sliced =
+            defined(sliced) ? sliced : !point.sliced;
+        // update userOptions.data
+        (series.options.data as any)[series.data.indexOf(point)] =
+            point.options;
+
+        if (point.graphic) {
+            point.graphic.animate(this.getTranslate());
+        }
+
+        if (point.shadowGroup) {
+            point.shadowGroup.animate(this.getTranslate());
+        }
+    }
+}
+
+/* *
+ *
+ *  Prototype Properties
+ *
+ * */
+
+interface PiePoint {
+    connectorShapes: Record<string, PiePoint.ConnectorShapeFunction>;
+    labelDistance: number;
+    name: string;
+    options: PiePointOptions;
+    series: PieSeries;
+}
+extend(PiePoint.prototype, {
```

Comment:
Why do we do extend in this strange manner instead of just putting it in the class?

Reply:
In the class they need a default value in the constructor. A later `init` call by chart is not sufficient.

Reply:
Also to keep compatibility, the properties in the `extend` call continue to be set on the prototype. In a class they would be set for each instance separately, which can break code, fiddles, tests, and wraps.

Reply:
I will improve the situation with `any` casts. It has to be refactored anyway later on.